---
output:
  html_document: default
  pdf_document: default
---
# Introduction {#intro}

Yeasts (Saccharomyces cerevisiae) are used in the production of some of the most cherished food choices (e.g. bakery, wine making, and beer brewing). There are many other biotechnology applications that use yeast such as pharmaceutical and biomass production. 

Yeasts are great model organisms because of their simple and small genome consisting of approximately 6000 genes. As single celled organisms, they also make great models for transcriptome analyses as gene expression is homogenous. 

As part of our Vancouver-based hackathon (hackseq19) project examined  yeast transcriptome data scraped off of the [web](https://github.com/rtwillett/yeastract_spider/). The data consists of gene expression changes from yeast strains that have been treated with various stimuli such as heat, phenol lysis, ethanol treatment, etc.  Gene expression was normalized to Transcript Per Million (TPM). 

## Data Source 
This project is inspired by the open source yeast-omics dataset shared as a Kaggle competition. The original data can be found [here](https://www.kaggle.com/costalaether/yeast-transcriptomics) and scraped off from [here](https://github.com/rtwillett/yeastract_spider/).

## Data
The data in this project includes gene expression values for 92 yeast strains treated with various stimuli. RNA expression levels are normalized to TPM (transcripts per million), following a default normalization procedure. Data is stored in `data` folder.
- The `SC_expression.csv` file contains gene expression of yeast strains in the experiments.
- The `labels.csv` files pertain to gene validation status and molecular function (MF), cellular component (CC), and biological processes (BP) of those genes. 
- The `conditions_annotation.csv` file explain the yeast strains and experimental conditions.

## Processed Data
Over the course of the hackathon, the raw data was processed, cleaned and then analyzed in several steps prior to being used in the Shiny application. All data processing steps have been collected in numbered directories in the order that the relevant processing was done.  

* `/01_fixlabels` - Removed rows with empty fields.

* `/02_remove_gene` - A row with a geneID of `gene` rather than an ID code was spotted during cleaning, and was removed.

* `/03_remove_zero_gene` - Removed leftover rows with empty fields.

* `/04_remove_underscores_average_replicates` - Rows with geneID's that started with underscore characters that represent uncategorized reads were found and removed.

* `/05_grouping_experiments` - Added an additional column that specifies tags for groups of geneID's that belong to the same experiment.

* `/06_get_GO_terms` - Added gene onthology terms.
