---
output:
  html_document: default
  pdf_document: default
---

```{r}
knitr::opts_chunk$set(eval = FALSE)
```

# Methods

## Cleaning data
### Removing zeros
```{r}
here::here()
```
Get file
```{r}
expression <- read.csv("02_expression.csv", header = T, row.names = 1)
```
See which rows have all zeros
```{r}
zeros <- apply(expression, 1, function(x) return(length(which(x == 0))))
```
Generate new data frame
```{r}
newExpression <- expression[which(zeros != ncol(expression)),]
```
Write to file
```{r}
write.csv(newExpression, "03_SC_expression.csv", row.names = T)
```

### Check Normalization - Matt Emery
```{r}
expressions <- read_csv(here("data", "tidy_data", "tidy_expressions.csv"))
```

```{r}
expressions %>% 
  group_by(condition) %>% 
  summarise(average = mean(value), standard_deviation = sd(value)) %>% 
  arrange(standard_deviation)
```

```{r}
standard_deviation_gene <- expressions %>% 
  group_by(gene) %>% 
  summarise(average = mean(value), standard_deviation = sd(value)) %>% 
  arrange(standard_deviation)
```

Expressed in at least 5 samples:
```{r}
standard_deviation_gene %>% 
  filter(!str_detect(gene, "^__")) %>% 
  filter(standard_deviation > 0) %>% 
  arrange(standard_deviation)
  
```

```{r}
standard_deviation_gene %>% 
  arrange(desc(standard_deviation))
```

### Remove Underscores and average replicates - Matt Emery
```{r}
expressions <- read_csv(here("data/03_remove_zero_genes/03_SC_expression.csv"))
```

```{r}
removed_underscores <- expressions %>%
  rename("gene" = "X1") %>% 
  pivot_longer(-gene) %>% 
  filter(!str_detect(gene, "^_"))
```

```{r}
averaged_replicates <- removed_underscores %>% 
  mutate(name = str_remove(name, ".1$")) %>% 
  group_by(gene, name) %>% 
  summarise(value = mean(value))
```

```{r}
wide_df <- averaged_replicates %>% 
  pivot_wider(id_cols = gene)
```

```{r}
write_csv(wide_df, here("data/04_remove_underscores_average_replicates/04_SC_expression.csv"))
```

###  Load the expression matrix and associated metadata. Fix inconsistency between.
Resave the cleaned tables.
```{r}
library(tidyverse)
```
expression matrix, note that first column is gene names. coercing to rownames
expect 6071x93
```{r}
SC_expression <- read.csv("data/SC_expression.csv")
colnames(SC_expression)[colnames(SC_expression) == "X"] <- "gene"
```
expect 92x4
```{r}
condition_annotation <- read.csv("data/conditions_annotation.csv", stringsAsFactors = FALSE)
```
expect 6071x3
```{r}
labels_BP <- read.csv("data/labelsBP.csv", stringsAsFactors = FALSE)
```

expect 6071x3
```{r}
labels_CC <- read.csv("data/labelsCC.csv", stringsAsFactors = FALSE)
```

expect 6071x3
```{r}
labels_MF <- read.csv(
  "data/labelsMF.csv",
  stringsAsFactors = FALSE,
  header = FALSE,
  col.names = c("gene", "validation", "localization")
)
```
metadata has 6072 rows instead of 6071. turns out there is an extra row
with a gene called "gene." must remove
```{r}
table(labels_MF$gene %in% rownames(SC_expression))
setdiff(labels_MF$gene, rownames(SC_expression))
```

```{r}
labels_BP <- labels_BP[labels_BP$gene != "gene", ]
labels_CC <- labels_BP[labels_CC$gene != "gene", ]
labels_MF <- labels_BP[labels_MF$gene != "gene", ]
```

```{r}
stopifnot(all(table(SC_expression$gene %in% labels_BP$gene)))
stopifnot(all(table(SC_expression$gene %in% labels_CC$gene)))
stopifnot(all(table(SC_expression$gene %in% labels_MF$gene)))
```

```{r}
table.names <- c("01_condition_annotation.csv", 
                 "01_labelsBP.csv",
                 "01_labelsCC.csv",
                 "01_labelsMF.csv", 
                 "01_SC_expression.csv")

tables <- list(condition_annotation,
            labels_BP,
            labels_CC,
            labels_MF,
            SC_expression)
```

```{r}
for (i in 1:length(tables)) {
  write.table(tables[i], 
              sep = ",",
              quote = FALSE,
              row.names = FALSE,
              paste0("data/", table.names[i]))
}
```

### Grouping experiments 
load package dependancy
```{r}
library(tidyverse)
library(stringr)
```

can download limma through the bioconductor:
```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("limma")
library(limma)
```

set time zone to proper location
```{r}
options(readr.default_locale=readr::locale(tz="America/Vancouver"))
```


Load datafiles
```{r}
conditions_annotation <- read_csv("data/00_raw/conditions_annotation.csv")
SC_expression <- read_csv("data/04_remove_underscores_average_replicates/04_SC_expression.csv")


View(conditions_annotation)
```
Create group based on first to char in ID which we believe groups data from the same lab

```{r}
conditions_annotation$Group <- NA
conditions_annotation$Group <- substr(conditions_annotation$ID, 1, 2)   
```
convert everything in primary and secondary to lower case as currently mixed case
```{r}
conditions_annotation$primary <- str_to_lower(conditions_annotation$primary, locale = "en")

conditions_annotation$secondary <- str_to_lower(conditions_annotation$secondary, locale = "en")

conditions_annotation$Ref <- NA
```

split out the SA group and merge in gene expression data

```{r}
group_SA <- conditions_annotation[conditions_annotation$Group == "SA", ]

SA_expression <- SC_expression$gene

saa_subset <- group_SA %>% 
  pull(ID)

SA_expressiontemp <- SC_expression %>% 
  select(saa_subset)

SAtable <- cbind(SA_expression, SA_expressiontemp)
colnames(SAtable)[1] <- "Gene"

SAwildGalactose <- SAwildGalactose[,1:3]

rownames(SAwildGalactose) <- SAwildGalactose$Gene

SAwildGalactose$Gene <- NULL
```

```{r}
fit <- lmFit(SAwildGalactose)

SAwildGalactose$fold <- SAwildGalactose$SAASAQ/SAwildGalactose$SAABRQ
SAwildGalactose$pval <- fit$p.value
```

```{r}
df <- SAwildGalactose[is.finite(rowSums(SAwildGalactose)),]

g = ggplot(data=df, aes(x=log2(fold), y=-log10(pval))) +
  geom_point(alpha=0.4, size=5) +
  theme(legend.position = "none") +
  xlim(c(-5, 12)) + ylim(c(0, 5)) +
  xlab("log2 fold change") + ylab("-log10 p-value")
```




## Shiny App


## Visualizations

### UMAP Generation - Matt Emery
```{r}
library(umapr)
library(tidyverse)
```

```{r}
expression <- read_csv(here("data/04_remove_underscores_average_replicates/04_SC_expression.csv"))
```

```{r}
expression_matrix <- expression %>% 
  select_if(is.numeric) %>% 
  as.matrix()
```


```{r}
umap_expression <- umap(expression_matrix)
```

```{r}
umap_df <- bind_cols(expression %>% select(gene), umap_expression %>% select(UMAP1, UMAP2))
```


```{r}
umap_df <- umap_df %>% 
  mutate(long_non_coding = str_detect(gene, "^SCER_lnc"))
```

```{r}
write_csv(umap_df, here("data/umap.csv"))
```

```{r}
ggplot(umap_df, aes(UMAP1, UMAP2, color = long_non_coding)) + 
  geom_point(alpha = 0.2)
```

```{r}
tidy_attributes <- read_csv(here("data/tidy_data/tidy_attributes.csv"))
```

```{r}
wide_attributes <- tidy_attributes %>%
  pivot_wider(gene)
```

```{r}
attributes_df <- umap_df %>% 
  inner_join(wide_attributes)
```

```{r}
umap_coloring <- function(filter_query) {
  filtered_values <- tidy_attributes %>% 
    filter(value == {{ filter_query }}) %>% 
    pull(gene)
  
  
  filter_df <- umap_df %>% 
    mutate({{ filter_query }} := map_lgl(gene, function(x) x %in% filtered_values))
  
  ggplot(filter_df, aes_string("UMAP1", "UMAP2", color = filter_query)) + 
    geom_point()
  }

g <- umap_coloring("protein") 
  ggsave(here("output/first_umap.png"), g, width = 16, height = 9)
```

```{r}
tidy_attributes %>% 
  distinct(value)
```

### Heat Map Generation

---

```{r libs, echo=F}
library(fs)
library(here)
library(tidyverse)
library(DESeq2)
library(Rtsne)
```

```{r read}

rel_expr <- read_csv(fs::path(here::here(), "data","00_raw","SC_expression.csv"))

strain_meta <- read_csv(fs::path(here::here(), "data","00_raw","conditions_annotation.csv"))

# read in the gene metadata, with additional cols for tags
tag.vec <- c("name","confidence", paste0("tag",1:10))
# biol process
bp_meta <- read_csv(fs::path(here::here(),"data","00_raw","labels_BP.csv"), col_names = tag.vec)
# cellular_component
cc_meta <- read_csv(fs::path(here::here(),"data","00_raw","labels_CC.csv"), col_names = tag.vec)
# molecular func
mf_meta <- read_csv(fs::path(here::here(),"data","00_raw","labels_MF.csv"), col_names = tag.vec)

```

```{r wrangle}
# cols are strains, rows are genes

# threshold <- 0.0001 # 
# remove dups
rel_expr <- rel_expr %>%
  select(-ends_with("_1")) %>% # remove duplicate strains
  dplyr::rename(gene_name = X1) %>% 
  pivot_longer(-gene_name, names_to = "culture_treatment", values_to = "rel_expr") %>% 
  select(culture_treatment, gene_name, rel_expr) #%>% # reorder just to make look better
  # filter(rel_expr > threshold)

rel_expr
```

```{r wrangle2}
strain_meta %>% distinct(primary)
strain_meta %>% distinct(secondary)

strain_meta <- strain_meta %>% 
  dplyr::rename(culture_treatment = ID) %>% 
  mutate(secondary = if_else(secondary == "<not provided>", NA_character_, secondary))
head(strain_meta) # not sure how to tidy this
```

```{r wrangle3}
clean_annotations <- function(df){
  df %>% 
    pivot_longer(cols = paste0("tag",1:10), names_to = "tag", values_to = "vals") %>% 
    drop_na() %>% 
    select(-tag) %>% 
    dplyr::rename(tag = vals,
                  gene_name = name) %>% 
    filter(gene_name != "gene",
           !tag %in% c("biological_process","molecular_function", "cellular_component"),
           confidence == "Verified") %>% # remove artifacts
    select(-confidence) 
}
bp_meta <- bp_meta %>%
  clean_annotations()
bp_meta %>% 
  distinct(tag)
cc_meta <- cc_meta %>% 
  clean_annotations()
cc_meta %>% 
  distinct(tag)
mf_meta <- mf_meta %>% 
  clean_annotations()
cc_meta %>% 
  distinct(tag)
```

```{r eda}
go_type <- mf_meta

# compare RNA expression of strains with go_tag1 and go_tag2
heatmap_by_GO <- rel_expr %>% 
  left_join(go_type, by="gene_name") %>%
  left_join(strain_meta, by="culture_treatment") %>% 
  # filter(tag %in% go_tags) %>% 
  group_by(tag, primary) %>% 
  summarise(rel_expr = mean(rel_expr)) %>% 
  ungroup()
  
  
ggplot(heatmap_by_GO, aes(x=primary %>% fct_reorder(-rel_expr), y=tag %>% fct_reorder(-rel_expr))) +
  geom_tile(aes(fill=rel_expr)) +
  scale_fill_viridis_c() +
  ggtitle("Mean transcript abundance") +
  theme(axis.text.x = element_text(angle = 90, hjust=0.99, vjust=0.5)) +
  ylab("Molecular Function (GO)") +
  xlab("primary") +
  labs(fill="Norm. rel. expr.")
  
```







